gym/wrappers/test_clip_action.py

```diff
@@ -0,0 +1,18 @@
+import gym
+from gym.wrappers import ClipAction
+
+
+def test_clip_action():
+    # mountaincar: action-based rewards
+    env = gym.make('MountainCarContinuous-v0')
+    clipped_env = ClipAction(env)
+
+    env.reset()
+    clipped_env.reset()
+
+    action = [10000.]
+
+    _, reward, _, _ = env.step(action)
+    _, clipped_reward, _, _ = clipped_env.step(action)
+
+    assert abs(clipped_reward) < abs(reward)
```

Comment:
this particular assertion seems to correlate rather weakly with action clipping (also wording clipped_reward seems confusing given that it is the action that is being clipped). If I may suggest an alternative - how about creating two environments and seeding them the same way, then passing an action at the action space boundary to the unwrapped, and outside of action space limits to wrapped, and making sure the results are identical?

Reply:
or maybe even create a sequence of actions, some of which can be outside action space boundary, clipping them, and passing clipped sequence to unwrapped env and unclipped sequence to a wrapped env and comparing the results?

Reply:
Hi @pzhokhov , thank you for the comment, it is a great idea ! I've updated the test function accordingly to the second proposal, such that we firstly seed the environments, and generate action sequence with some of the actions out of the boundary, and passing clipped action to original environment and unclipped to the wrapped environment with their rewards compared.

Reply:
I am still not satisfied :) If I understand correctly, current test is actually taking the same steps twice on the same environment (because clipped_env.step() calls env.step()) - instead of taking same steps of two *copies* of the environment. The reason why rewards match is because the mountain car produces -1 reward on every step :). Ideally, the *observations* should match

Reply:
Indeed, now new assertion comparing observations is added.